### PR TITLE
[#253] Add access box cache

### LINKED ISSUE #253: Cache accessbox object
According to pprof, accessbox object is heavily fetched from the NeoFS during the single client workload. We can easily cache it and reuse for the future operations.

image

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #253 

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

**Comment by KirillovDenis** [comment]
before 
image

after
image

File: creds/tokens/accessbox_cache.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	if !ok {
+		return nil
+	}
+
+	return result
+}
+
+// Put stores an object to cache.
+func (o *AccessBoxCache) Put(address *object.Address, box *accessbox.Box) error {
+	return o.cache.Set(address.String(), box)
```
Comment (realloc) [inline creds/tokens/accessbox_cache.go:40]: Maybe it worth setting the expiration matching the AccessBox's expiration time.

File: creds/tokens/accessbox_cache.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	"github.com/bluele/gcache"
+	"github.com/nspcc-dev/neofs-api-go/pkg/object"
+	"github.com/nspcc-dev/neofs-s3-gw/creds/accessbox"
+)
+
+const (
+	// DefaultCacheSize is a default lifetime of entries in  cache.
+	DefaultCacheSize = 100
+	// DefaultCacheLifetime is a default maximum number of entries in cache.
```
Comment (realloc) [inline creds/tokens/accessbox_cache.go:14]: Should be vise-versa =)

Comment (KirillovDenis) [reply creds/tokens/accessbox_cache.go]: Fixed

**Review by alexvanin — APPROVED** [review]

**Pull request merged**